src/checks/stylelint.ts

```diff
@@ -0,0 +1,42 @@
+import { execaCommand } from 'execa';
+import { LintResult } from 'stylelint';
+
+export const supportedExtensions = ['css', 'scss', 'less', 'js', 'tsx', 'jsx'];
+
+export const title = 'Stylelint';
+
+export default async function stylelintCheck() {
+  try {
+    await execaCommand(
+      `yarn stylelint **/*.{${supportedExtensions.join(
+        ',',
+      )}} --max-warnings 0 --formatter json`,
+    );
+  } catch (error) {
+    const { stdout } = error as { stdout: string };
+
+    const stylelintJSONOutput = stdout.slice(
+      stdout.indexOf('['),
+      stdout.lastIndexOf(']') + 1,
+    );
+
+    // If no Stylelint errors detected, throw the error
+    if (!/"errored":/.test(stylelintJSONOutput)) {
+      throw error;
+    }
+
+    const stylelintErrors = (
+      JSON.parse(stylelintJSONOutput) as LintResult[]
+    ).filter((stylelintError) => stylelintError.errored);
```

Comment:
In reality this represents the result of stylelint check in a single file and not neccessarily an error.

maybe stylelintResult or lintResult would be more accurate

Reply:
8894a6f